tools/packagehallucination/ruby/main.py

```diff
@@ -0,0 +1,87 @@
+import time
+import requests
+from datetime import datetime, date
+import backoff
+
+
+@backoff.on_exception(backoff.expo,
+                      (requests.exceptions.RequestException, requests.exceptions.HTTPError),
+                      max_tries=5)
+def get_gem_first_push_date(gem_name):
+    url = f"https://rubygems.org/api/v1/versions/{gem_name}.json"
+    response = requests.get(url, timeout=30)
+    response.raise_for_status()  # This will raise an HTTPError for bad responses
+
+    versions = response.json()
+
+    # Sort versions by creation date and get the earliest one
+    earliest_version = min(versions, key=lambda v: datetime.strptime(v['created_at'], "%Y-%m-%dT%H:%M:%S.%fZ"))
+    
+    first_push_date = datetime.strptime(earliest_version['created_at'], "%Y-%m-%dT%H:%M:%S.%fZ").date()
+    
+    return first_push_date
+
+def main():
+    cutoff_date = date(2023, 3, 1)
```

Comment:
Would prefer to keep the date in the dataset, so that the cutoff can be dynamically set in the detector component. I.e. the datasets should be comprehensive and unconstrained.

Reply:
@leondz  removed the cutoff date from script ✅ . Will re-run script and update dataset on huggingface